File: javascript/leetcode/custom-sort-string.js
/**
 * @param {string} S
 * @param {string} T
 * @return {string}
 */
var customSortString = function(S, T) {
    let l = T.split('');
    let r = [];
    S.split('').forEach(function (c) {
        let findIndex;
        while (findIndex !== -1) {
            findIndex = l.indexOf(c)
            if (findIndex !== -1) {
                r.push(l.splice(findIndex, 1))
            }
        }
    });
    return (r.concat(l)).join('');
};

File: python/leetcode/custom-sort-string.py
class Solution:
    def customSortString(self, S, T):
        """
        :type S: str
        :type T: str
        :rtype: str
        """
        l = list(T)
        r = []
        for c in S:
            findIndex = -1
            while findIndex is not None:
                try:
                    findIndex = l.index(c)
                except:
                    findIndex = None
                if findIndex is not None and findIndex is not -1:
                    r.append(l.pop(findIndex))
        return ''.join(r + l)
